### DISCUSSION TITLE
Some Questions about Application-Level Service Discovery in Dubbo 3.x

### DISCUSSION THREAD

**Discussion by jyyfei** [apache/dubbo#14221] [Question] [answered]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

Prerequisite:
Using pure application-level service discovery, the DubboReference annotation is used to implement the consumer. The consumer first goes to the metadata center to find the configuration of the Group as mapping to obtain the application to which the interface belongs, and then obtains the application instance IP from the registry according to the application.

Question:
If an interface was initially in application A and later migrated to application B, but the mapping configuration in the metadata center still includes both applications A and B, how can we ensure that the service is correctly invoked in application B? Will there be an additional layer of validation for the interfaces that actually exist in applications A and B?

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by chickenlj** [comment]
> If an interface was initially in application A and later migrated to application B, but the mapping configuration in the metadata center still includes both applications A and B, how can we ensure that the service is correctly invoked in application B? Will there be an additional layer of validation for the interfaces that actually exist in applications A and B?

See Consumer relies on service DemoService, Consumer will consume both A and B as long as there are available instances for A and B and both A and B export DemoService. This guarantees smooth migration. The only way for Consumer to only consumes B is to let A not export DemoService, Consumer is able to find that and send all traffics to B.

**Reply by jyyfei** [reply]
1、How do consumers obtain the provider application only through the interface? Is it through the metadata configuration whose group is "mapping"？

2、group's metadata configuration for "mapping" is probably outdated and not accurate, right? So it is necessary to further filter out the instances that do not export this interface through service Instance？

**Comment by chickenlj** [comment, marked as answer]
> 1、How do consumers obtain the provider application only through the interface? Is it through the metadata configuration whose group is "mapping"？

Yes

> 
> 2、group's metadata configuration for "mapping" is probably outdated and not accurate, right? So it is necessary to further filter out the instances that do not export this interface through service Instance？

Yes, consumer will do that and find out the instances that provides the service.
